client/src/components/comparison/Comparison.jsx

```diff
@@ -1,9 +1,100 @@
-import React from 'react';
+import React, {useEffect, useState/*, useContext*/} from 'react';
+import RelatedProducts from './RelatedProducts.jsx';
+import UserOutfitItems from './UserOutfitItems.jsx';
+// import {products} from '../../../mock-data/products.js';
+// import {productStyles} from '../../../mock-data/productStyles.js';
+import Requester from '../../Requester.js';
+import LoadingSpinner from '../../components/shared/LoadingSpinner.jsx';
 
-const Comparison = (props) => {
-  return (
-    <div>Comparison</div>
-  );
-};
+const rerequester = Requester();
+
+class Comparison extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      relatedProducts: this.props.relatedProducts,
+      detaledRelatedProductsArr: [],
+      relatedProductsStylesArr: [],
+      outfitList: [],
+      reviewsMetaArr: [],
+      activeIndex: 0,
+      styles: this.props.styles
+    };
+  }
+
+  // function that will handle product click
+
+  // function that will handle the compare Modal
+
+  // function that will handle adding an item to the outfit list
+
+  // handleAddOutfitItem = (product, style) => {
+  //   if (!outfitList.some(obj => obj.id === product.id)) {
+  //     this.setState({
+  //       outfitList: [...outfitList, product]});
+  //   }
+  // };
+
+  // function that will handle removing an item form the outfit list
+
+  // get and store the product obj from each product in the relatedProducts array
+  getTheDeets = ()=> {
+    return Promise.all(this.state.relatedProducts.map(relatedProduct => rerequester.getProduct(relatedProduct)))
+      .then((data) => {
+        // console.log('this is working', data);
+        this.setState({
+          detaledRelatedProductsArr: data
+        });
+      })
+      .then(this.getTheStyles())
```

Comment:
Does this work? Adding the parens after `getTheStyles` seems like it would invoke the function immediately. I think (not 100% sure here) it should be `.then(this.getTheStyles)`.

Reply:
.then(() => {
        return this.getTheStyles();
      })
I changed it to look like the above, I could make it slightly smaller but i find this a little easier to read, however, I can change it to .then(this.getTheStyles) if you would prefer.